kptncook/mealie.py

```diff
@@ -187,51 +172,265 @@ def login(self, username: str = "admin", password: str = ""):
         r = self.put(recipe_detail_path, data=recipe.json())
+        r.close()
         r.raise_for_status()
         return Recipe.parse_obj(r.json())
 
     def create_recipe(self, recipe):
-        slug = self._post_recipe_and_get_slug(recipe)
+        slug = self._post_recipe_trunk_and_get_slug(recipe.name)
+        print(slug)
+        recipe.slug = slug
         self._scrape_image_for_recipe(recipe, slug)
         recipe = self._update_user_and_group_id(recipe, slug)
+        recipe = self.enrich_recipe_with_step_images(recipe)
         return self._update_recipe(recipe, slug)
 
     def get_all_recipes(self):
-        r = self.get("/recipes")
+        all_recipes = []
+
+        r = self.get("/recipes?page=1&perPage=50")
+        r.close()
         r.raise_for_status()
-        return [Recipe.parse_obj(recipe) for recipe in r.json()]
+        all_recipes.extend([Recipe.parse_obj(recipe) for recipe in r.json()["items"]])
+
+        page = 2
+        while page <= r.json()["total_pages"]:
+            r = self.get(f"/recipes?page={page}&perPage=50")
+            r.close()
+            r.raise_for_status()
+            all_recipes.extend([Recipe.parse_obj(recipe) for recipe in r.json()["items"]])
+            page += 1
+            
+        return all_recipes
 
     def delete_via_slug(self, slug):
         r = self.delete(f"/recipes/{slug}")
+        r.close()
         r.raise_for_status()
         return r.json()
 
     def get_via_slug(self, slug):
         r = self.get(f"/recipes/{slug}")
+        r.close()
         r.raise_for_status()
         return Recipe.parse_obj(r.json())
+
+
+def kptncook_to_mealie(kcin: KptnCookRecipe, mealie_client: MealieApiClient, api_key: str = settings.kptncook_api_key) -> RecipeWithImage:
+    kwargs = {
+        "name": kcin.localized_title.de,
+        "notes": [
+            RecipeNote(title="author comment", text=kcin.author_comment.de),
+        ],
+        "nutrition": Nutrition(
+            calories=kcin.recipe_nutrition.calories,
+            proteinContent=kcin.recipe_nutrition.protein,
+            fatContent=kcin.recipe_nutrition.fat,
+            carbohydrateContent=kcin.recipe_nutrition.carbohydrate,
+        ),  # type: ignore
+        "prepTime": kcin.preparation_time,
+        "cookTime": kcin.cooking_time,
+        "recipeYield": "1 Portionen", # kptncook serves for default 2 portions, but we want it more granular
+        "recipeInstructions": [
+            RecipeStep(title=None, text=step.title.de, image=step.image) for step in kcin.steps
+        ],
+        "recipeIngredient": [
+            RecipeIngredient(food=mealie_client.get_food_by_name(ig.ingredient.localized_title.de.split(",")[0]), quantity=(ig.quantity/2.0) if ig.quantity is not None else None, unit=mealie_client.get_unit_by_name(ig.measure if hasattr(ig, "measure") else None), note=ig.ingredient.localized_title.de.split(",")[1] if len(ig.ingredient.localized_title.de.split(","))>1 else None)  # type: ignore
```

Comment:
Whoa! 😅

Reply:
Sorry for that 😬 The line has grown over the course of 3 weeks 😅 Feel free to simplify, my first goal was to get it working